### PR TITLE
Notebook and CLI examples for DinoV2 models for embedding

### PR DESCRIPTION
**Pull request by rdondera-microsoft**
# Description

Notebooks and CLI examples for online and batch deployment for DinoV2 base for image embedding tasks.

Tested by running all four combinations:
a. notebook-online
notebook_online2
b. notebook-batch: https://ml.azure.com/runs/batchjob-000c2e50-766a-4e81-a166-b10c4386117f?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/llava&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#
c. CLI-online: failed due to error not related to current changes ("No valid deployments to route to. Please check that the endpoint has at least one deployment with positive weight values or use a deployment specific header to route.")
d. CLI-batch: https://ml.azure.com/runs/batchjob-3a3343cf-1ffc-4319-af1c-94cd28ade1a9?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rdondera/providers/Microsoft.MachineLearningServices/workspaces/llava&tid=72f988bf-86f1-41af-91ab-2d7cd011db47

# Checklist

- [x] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [x] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [x] Pull request includes test coverage for the included changes.
- [x] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+   "source": [
+    "## Image Embeddings Inference using Batch Endpoints\n",
+    "\n",
+    "This sample notebook shows how to deploy `embeddings` type models to a batch endpoint for image embeddings inference.\n",
+    "\n",
+    "### Task\n",
+    "The input to the `embeddings` task is images, text or image-text pairs, and the output is feature embeddings produced by the model. In this notebook, we are only concerned with image inputs.\n",
+    " \n",
+    "### Model\n",
+    "Models that can perform the `embeddings` task are tagged with `embeddings`. We will use the `Facebook-DinoV2-Image-Embeddings-ViT-Base` model in this notebook. If you opened this notebook from a specific model card, remember to replace the specific model name. If you don't find a model that suits your scenario or domain, you can discover and [import models from HuggingFace hub](../../import/import_model_into_registry.ipynb) and then use them for inference. \n",
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb:15]: nit: may be you can mention about model in catalog.

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    "\n",
+    "request_json = {\n",
+    "    \"input_data\": {\n",
+    "        \"columns\": [\"image\", \"text\"],\n",
+    "        \"index\": [0, 1],\n",
+    "        \"data\": [\n",
+    "            [\n",
+    "                base64.encodebytes(read_image(sample_image_1)).decode(\"utf-8\"),\n",
+    "                \"\",\n",
+    "            ],  # the \"text\" column should contain empty string\n",
```
Comment (gauravrajguru) [inline sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb:284]: nit: we can elaborate little more, like, how to send text only, image-text to deployment to get their embeddings.

**Review by gauravrajguru — APPROVED** [review]
nit comments.

**Comment by nvijayrania** [comment]
Is the cli-online endpoint failing on master as well?

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb (hunk 1 above)
Comment (rdondera-microsoft) [reply sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb:15]: Done.

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb (hunk 2 above)
Comment (rdondera-microsoft) [reply sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb:284]: DinoV2 is for images only and the reference to text is a mistake. Thanks for catching!

**Comment by rdondera-microsoft** [comment]
> Is the cli-online endpoint failing on master as well for other tasks such as image-classification? Wondering if it's customer impacting already

Turns out it does, unless my setup for running CLI examples is wrong. Let's sync on this, but in the meantime consider approving this PR since the infrastructure issue is not specific to it.

**Review by nvijayrania — CHANGES_REQUESTED** [review]

File: cli/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
+set -x
+# The commands in this file map to steps in this notebook: https://aka.ms/azureml-infer-batch-sdk-image-classification
```
[the remark this answers is not in the document]
Comment (rdondera-microsoft) [reply cli/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.sh:2]: Yes. Changed to https://aka.ms/azureml-infer-cli-image-embeddings.

File: cli/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
+set -x
+# The commands in this file map to steps in this notebook: <TODO>
```
[the remark this answers is not in the document]
Comment (rdondera-microsoft) [reply cli/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.sh:2]: Link added.

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workspace_ml_client.batch_endpoints.begin_delete(name=endpoint_name).result()"
+   ]
+  }
+ ],
+ "metadata": {
```
[the remark this answers is not in the document]
Comment (rdondera-microsoft) [reply sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-batch-endpoint.ipynb:473]: Done.

File: sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workspace_ml_client.online_endpoints.begin_delete(name=online_endpoint_name).wait()"
+   ]
+  }
+ ],
+ "metadata": {
```
[the remark this answers is not in the document]
Comment (rdondera-microsoft) [reply sdk/python/foundation-models/system/inference/image-embeddings/image-embeddings-online-endpoint.ipynb:327]: Done.

**Review by nvijayrania — APPROVED** [review]

**Pull request merged**